staging/src/k8s.io/component-base/cli/flag/string_slice_flag.go

```diff
@@ -46,8 +46,7 @@ func (s *StringSlice) String() string {
 
 func (s *StringSlice) Set(val string) error {
 	if s.value == nil || !s.changed {
-		v := make([]string, 0)
-		s.value = &v
```

Comment:
`
func NewStringSlice(s *[]string) *StringSlice {
	return &StringSlice{value: s}
}
`
`s.value = &v`, this makes it use a new pointer, which obviously shouldn't be done

Reply:
does 'flags.VarP()'  work?

Reply:
> does 'flags.VarP()' work?

https://github.com/kubernetes/kubernetes/blob/c9414c2bbaf9acfc245fc481ba2831efe7fdc7d6/staging/src/k8s.io/component-base/cli/flag/string_slice_flag_test.go#L69

 After change to `fs.VarP(v, "slice", "", "usage")`,  if the expected compared with `s`,  the test still fails

Reply:
`*s.value = make([]string, 0)` will panic if  s.Value is nil

Reply:
Do you mean this issues https://github.com/kubernetes/kubernetes/issues/100342?

`nil` judgment is added to this pr https://github.com/kubernetes/kubernetes/pull/100362 , so there will be no such problem.
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/component-base/cli/flag/string_slice_flag.go#L41

Retest without problems.
```
make WHAT=test/e2e/e2e.test
./_output/bin/e2e.test --help
```

Reply:
how about change like that

```
func (s *StringSlice) Set(val string) error {
	if s.value == nil {
		return fmt.Errorf("return err")
	}
	if *s.value == nil || !s.changed {
		*s.value = make([]string, 0)
	}
	*s.value = append(*s.value, val)
	s.changed = true
	return nil
}
```

Reply:
> Do you mean this issues #100342?
> 
> `nil` judgment is added to this pr #100362 , so there will be no such problem.
> https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/component-base/cli/flag/string_slice_flag.go#L41
> 
> Retest without problems.
> 
> ```
> make WHAT=test/e2e/e2e.test
> ./_output/bin/e2e.test --help
> ```

yes

Reply:
> how about change like that
> 
> ```
> func (s *StringSlice) Set(val string) error {
> 	if s.value == nil {
> 		return fmt.Errorf("return err")
> 	}
> 	if *s.value == nil || !s.changed {
> 		*s.value = make([]string, 0)
> 	}
> 	*s.value = append(*s.value, val)
> 	s.changed = true
> 	return nil
> }
> ```

There is already a nil judgment before, is this necessary? In fact, the original repaired pr only needs to be judged by nil, what I did is to restore the redundant changes. As you can see, one of the kind of pr is `regression`

Reply:
no, just  consider the Robustness of `Set` function, 
```
if err := NewStringSlice(nil).Set("value"); err != nil {
		t.Errorf("set value error: %v", err)
	}
````
the error is panic

Reply:
no, just  consider the Robustness of `Set` function, 
```
if err := NewStringSlice(nil).Set("value"); err != nil {
		t.Errorf("set value error: %v", err)
	}
````
the error is panic

Reply:
It looks good, but I think the role of StringSlice is to assign values to the input parameters, and it seems meaningless to pass nil.
:smile:

Reply:
> how about change like that
> 
> ```
> func (s *StringSlice) Set(val string) error {
> 	if s.value == nil {
> 		return fmt.Errorf("return err")
> 	}
> 	if *s.value == nil || !s.changed {
> 		*s.value = make([]string, 0)
> 	}
> 	*s.value = append(*s.value, val)
> 	s.changed = true
> 	return nil
> }
> ```

Modified and submitted